### ISSUE TITLE
Getting "x86/lib/fips_premain.c fingerprint mismatch" error

### ISSUE THREAD

**Issue by ware16** [openssl/openssl#15136]
I am trying to compile my application with x86 openSSL library (FIPS capable) for Android using Mac machine.

When I tried first, it was giving some permission error
```
/Users/x/Library/Android/ndk-r18b/sources/cxx-stl/llvm-libc++/libs/x86/libc++_shared.so" && :
  /Users/x/Code/downloaded/android/openssl/x86/fipsld: line 101: /Users/x/Code/downloaded/android/openssl/x86/lib/fips_premain.c: Permission denied
  1d0
  < HMAC-SHA1(fips_premain.c)= 2bfb57ef540bdba29220a45d65e1b4080de9adc1
  /Users/x/Code/downloaded/android/openssl/x86/lib/fips_premain.c fingerprint mismatch
  ninja: build stopped: subcommand failed.
```

Then I have given 755 permission to fips_premain.c and getting following error:
```
"/Users/x/Library/Android/ndk-r18b/sources/cxx-stl/llvm-libc++/libs/x86/libc++_shared.so" && :
  /Users/x/Code/downloaded/android/openssl/x86/lib/fips_premain.c: line 1: /Applications: is a directory
  /Users/x/Code/downloaded/android/openssl/x86/lib/fips_premain.c: line 2: syntax error near unexpected token `('
  /Users/x/Code/downloaded/android/openssl/x86/lib/fips_premain.c: line 2: ` * Copyright (c) 2005 The OpenSSL Project. Rights for redistribution'
  1d0
  < HMAC-SHA1(fips_premain.c)= 2bfb57ef540bdba29220a45d65e1b4080de9adc1
  /Users/x/Code/downloaded/android/openssl/x86/lib/fips_premain.c fingerprint mismatch
```

I am setting 
- FIPS_SIG = /Users/x/Code/downloaded/android/openssl/x86/incore
- FIPSLIBDIR = /Users/x/Code/downloaded/android/openssl/x86/lib
- FIPSDIR =  /Users/x/Code/downloaded/android/openssl/x86/fips
- FIPSLD_CC = /Users/x/Library/Android/ndk-r18b/toolchains/llvm/prebuilt/darwin-x86_64/bin/clang++

I have tried FIPS capable with and without 'shared' option.

Can you please help to understand what could be the root?

Thanks!

**Comment by paulidale** [comment]
This version of OpenSSL and the FIPS object module are no longer supported, however `fingerprint mismatch` almost certainly means what it says: a checksum is wrong.

**Comment by levitte** [comment]
```
  /Users/x/Code/downloaded/android/openssl/x86/lib/fips_premain.c: line 1: /Applications: is a directory
```

Is there a directory with a space involved somewhere, by any chance?

**Comment by nhorman** [comment]
Marking as inactive, to be closed at the end of 3.4 dev barring further input

**Comment by paulidale** [comment]
Long out of support, closing
